contracts/meta-oracles/LinearizedPriceDataSource.sol

```diff
@@ -0,0 +1,199 @@
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import { IMedian } from "../external/DappHub/interfaces/IMedian.sol";
+import { IDataFeed } from "./interfaces/IDataFeed.sol";
+
+
+/**
+ * @title LinearizedPriceDataSource
+ * @author Set Protocol
+ *
+ * This DataSource returns the current value of the Medianizer Oracle. If the interpolationThreshold
+ * is reached, then returns a linearly interpolated value.
+ * It is intended to be read by a DataFeed smart contract.
+ */
+contract LinearizedPriceDataSource is
+    Ownable
+{
+    using SafeMath for uint256;
+
+    /* ============ State Variables ============ */
+    uint256 public interpolationThreshold; 
+    string public dataDescription;
+    IMedian public medianizerInstance;
+
+    /* ============ Events ============ */
+
+    event LogMedianizerUpdated(
+        address newMedianizerAddress
+    );
+
+    /* ============ Constructor ============ */
+
+    /*
+     * Set interpolationThreshold, data description, and instantiate medianizer
+     *
+     * @param  _interpolationThreshold    The minimum time in seconds where interpolation is enabled
+     * @param  _medianizerAddress         The address to read current data from
+     * @param  _dataDescription           Description of contract for Etherscan / other applications
+     */
+    constructor(
+        uint256 _interpolationThreshold,
+        address _medianizerAddress,
+        string memory _dataDescription
+    )
+        public
+    {
+        interpolationThreshold = _interpolationThreshold;
+        medianizerInstance = IMedian(_medianizerAddress);
+        dataDescription = _dataDescription;
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * Returns the data from the Medianizer contract. If the current timestamp has surpassed
+     * the interpolationThreshold, then the current price is retrieved and interpolated based on
+     * the previous value and the time that has elapsed since the intended update value.
+     * Note: The sender must be a DataSource contract.
+     *
+     * Returns with newest data point by querying medianizer. Is eligible to be
+     * called after nextAvailableUpdate timestamp has passed. Because the nextAvailableUpdate occurs
```

Comment:
nextEarliestUpdate...but feels like this javadoc talking about restrictions on time should be put on the DataFeed poke function (as well as the underlying logic).

Reply:
I think we need to rework the javadocs here but that's about it. The stuff regarding the `nextEarliestUpdate` timestamp should probably be included in some form on the `DataFeed` contract if it isn't already. In my mind that includes everything after `Is eligible...` on line 80.